java/server/src/org/openqa/grid/common/RegistrationRequest.java

```diff
@@ -258,21 +133,15 @@ public static RegistrationRequest getNewInstance(String json) {
     try {
       JsonObject o = new JsonParser().parse(json).getAsJsonObject();
 
-      if (o.has("id")) request.setId(o.get("id").getAsString());
       if (o.has("name")) request.setName(o.get("name").getAsString());
       if (o.has("description")) request.setDescription(o.get("description").getAsString());
 
       JsonObject config = o.get("configuration").getAsJsonObject();
-      Map<String, Object> configuration = new JsonToBeanConverter().convert(Map.class, config);
-      // For backward compatibility numbers should be converted to integers
-      for (String key : configuration.keySet()) {
-        Object value = configuration.get(key);
-        if (value instanceof Long) {
-          configuration.put(key, ((Long) value).intValue());
-        }
-      }
+      GridNodeConfiguration configuration = new JsonToBeanConverter().convert(GridNodeConfiguration.class, config);
       request.setConfiguration(configuration);
 
+      if (o.has("id")) request.configuration.id = o.get("id").getAsString();
```

Comment:
"id" is another mess to cleanup... it's used interchangeably in both the configuration and in the registration object and in the 'main' block of a json formatted configuration file (which then has a 'configuration' section).

I'll add braces around blocks everywhere (i know there's some other spots like that)